contracts/cw721-sellable/src/execute.rs

```diff
@@ -101,6 +100,54 @@ pub fn try_list(
     Ok(Response::new().add_attribute("method", "list"))
 }
 
+pub fn try_redeem(
+    deps: DepsMut,
+    info: MessageInfo,
+    address: String,
+    ticket_id: &String,
+) -> Result<Response, ContractError> {
+    let contract = Cw721SellableContract::default();
+
+    // Validate only contract owner can call method
+    let minter = contract.minter.load(deps.storage)?;
+    if info.sender != minter {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Load ticket, error if ticket does not exist
+    let mut ticket = contract.tokens.load(deps.storage, ticket_id)?;
+
+    // Make sure owner param matches ticket owner
+    if ticket.owner != address {
+        return Err(ContractError::Unauthorized);
+    }
+
+    // Make sure ticket isn't locked or redeemed
+    match ticket.extension {
+        Some(ref metadata) => {
+            if (*metadata).redeemed.unwrap_or(false) {
+                return Err(ContractError::Redeemed);
+            } else if (*metadata).locked.unwrap_or(false) {
+                return Err(ContractError::Locked);
+            }
+        }
+        None => (),
+    }
+
+    // Mark ticket as redeemed and locked
+    match ticket.extension {
```

Comment:
Similarly, you can hoist this bit into the outer `if let` above, since you will have the metadata (you can make the ref above a ref mut (shorthand `&mut foo`). In fact, the else branch of the outer `if let Some(metadata) = ...` should be to raise an error `NoMetadataPresent` or something like that!